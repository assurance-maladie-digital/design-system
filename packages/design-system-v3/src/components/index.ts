export { default as Demo } from './Demo/Demo.vue';
