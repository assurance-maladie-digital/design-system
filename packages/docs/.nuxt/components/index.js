import { wrapFunctional } from './utils'

export { default as DocAlert } from '../../src/components/global/DocAlert.vue'
export { default as DocApi } from '../../src/components/global/DocApi.vue'
export { default as DocApiTable } from '../../src/components/global/DocApiTable.vue'
export { default as DocContentExplorer } from '../../src/components/global/DocContentExplorer.vue'
export { default as DocExample } from '../../src/components/global/DocExample.vue'
export { default as DocImage } from '../../src/components/global/DocImage.vue'
export { default as DocMarkup } from '../../src/components/global/DocMarkup.vue'
export { default as DocMaterialLoader } from '../../src/components/global/DocMaterialLoader.vue'
export { default as DocTabItem } from '../../src/components/global/DocTabItem.vue'
export { default as DocTabs } from '../../src/components/global/DocTabs.vue'
export { default as DocUsage } from '../../src/components/global/DocUsage.vue'
export { default as VueFile } from '../../src/components/global/VueFile.vue'
export { default as DocHeader } from '../../src/components/DocHeader.vue'
export { default as DocSearchField } from '../../src/components/DocSearchField.vue'
export { default as ContentExplorer } from '../../src/components/content-explorer/ContentExplorer.vue'
export { default as DrawerDocDrawer } from '../../src/components/drawer/DocDrawer.vue'
export { default as DrawerDocDrawerList } from '../../src/components/drawer/DocDrawerList.vue'
export { default as DrawerDocDrawerListGroup } from '../../src/components/drawer/DocDrawerListGroup.vue'
export { default as DrawerDocDrawerListItem } from '../../src/components/drawer/DocDrawerListItem.vue'
export { default as CodeDocCodeBlock } from '../../src/components/code/DocCodeBlock.vue'
export { default as CodeDocMarkdown } from '../../src/components/code/DocMarkdown.vue'
export { default as PageDocGithubLink } from '../../src/components/page/DocGithubLink.vue'
export { default as PageDocPageInfo } from '../../src/components/page/DocPageInfo.vue'
export { default as PageDocPrevNext } from '../../src/components/page/DocPrevNext.vue'
export { default as PageDocTooltipBtn } from '../../src/components/page/DocTooltipBtn.vue'

export const LazyDocAlert = import('../../src/components/global/DocAlert.vue' /* webpackChunkName: "components/doc-alert" */).then(c => wrapFunctional(c.default || c))
export const LazyDocApi = import('../../src/components/global/DocApi.vue' /* webpackChunkName: "components/doc-api" */).then(c => wrapFunctional(c.default || c))
export const LazyDocApiTable = import('../../src/components/global/DocApiTable.vue' /* webpackChunkName: "components/doc-api-table" */).then(c => wrapFunctional(c.default || c))
export const LazyDocContentExplorer = import('../../src/components/global/DocContentExplorer.vue' /* webpackChunkName: "components/doc-content-explorer" */).then(c => wrapFunctional(c.default || c))
export const LazyDocExample = import('../../src/components/global/DocExample.vue' /* webpackChunkName: "components/doc-example" */).then(c => wrapFunctional(c.default || c))
export const LazyDocImage = import('../../src/components/global/DocImage.vue' /* webpackChunkName: "components/doc-image" */).then(c => wrapFunctional(c.default || c))
export const LazyDocMarkup = import('../../src/components/global/DocMarkup.vue' /* webpackChunkName: "components/doc-markup" */).then(c => wrapFunctional(c.default || c))
export const LazyDocMaterialLoader = import('../../src/components/global/DocMaterialLoader.vue' /* webpackChunkName: "components/doc-material-loader" */).then(c => wrapFunctional(c.default || c))
export const LazyDocTabItem = import('../../src/components/global/DocTabItem.vue' /* webpackChunkName: "components/doc-tab-item" */).then(c => wrapFunctional(c.default || c))
export const LazyDocTabs = import('../../src/components/global/DocTabs.vue' /* webpackChunkName: "components/doc-tabs" */).then(c => wrapFunctional(c.default || c))
export const LazyDocUsage = import('../../src/components/global/DocUsage.vue' /* webpackChunkName: "components/doc-usage" */).then(c => wrapFunctional(c.default || c))
export const LazyVueFile = import('../../src/components/global/VueFile.vue' /* webpackChunkName: "components/vue-file" */).then(c => wrapFunctional(c.default || c))
export const LazyDocHeader = import('../../src/components/DocHeader.vue' /* webpackChunkName: "components/doc-header" */).then(c => wrapFunctional(c.default || c))
export const LazyDocSearchField = import('../../src/components/DocSearchField.vue' /* webpackChunkName: "components/doc-search-field" */).then(c => wrapFunctional(c.default || c))
export const LazyContentExplorer = import('../../src/components/content-explorer/ContentExplorer.vue' /* webpackChunkName: "components/content-explorer" */).then(c => wrapFunctional(c.default || c))
export const LazyDrawerDocDrawer = import('../../src/components/drawer/DocDrawer.vue' /* webpackChunkName: "components/drawer-doc-drawer" */).then(c => wrapFunctional(c.default || c))
export const LazyDrawerDocDrawerList = import('../../src/components/drawer/DocDrawerList.vue' /* webpackChunkName: "components/drawer-doc-drawer-list" */).then(c => wrapFunctional(c.default || c))
export const LazyDrawerDocDrawerListGroup = import('../../src/components/drawer/DocDrawerListGroup.vue' /* webpackChunkName: "components/drawer-doc-drawer-list-group" */).then(c => wrapFunctional(c.default || c))
export const LazyDrawerDocDrawerListItem = import('../../src/components/drawer/DocDrawerListItem.vue' /* webpackChunkName: "components/drawer-doc-drawer-list-item" */).then(c => wrapFunctional(c.default || c))
export const LazyCodeDocCodeBlock = import('../../src/components/code/DocCodeBlock.vue' /* webpackChunkName: "components/code-doc-code-block" */).then(c => wrapFunctional(c.default || c))
export const LazyCodeDocMarkdown = import('../../src/components/code/DocMarkdown.vue' /* webpackChunkName: "components/code-doc-markdown" */).then(c => wrapFunctional(c.default || c))
export const LazyPageDocGithubLink = import('../../src/components/page/DocGithubLink.vue' /* webpackChunkName: "components/page-doc-github-link" */).then(c => wrapFunctional(c.default || c))
export const LazyPageDocPageInfo = import('../../src/components/page/DocPageInfo.vue' /* webpackChunkName: "components/page-doc-page-info" */).then(c => wrapFunctional(c.default || c))
export const LazyPageDocPrevNext = import('../../src/components/page/DocPrevNext.vue' /* webpackChunkName: "components/page-doc-prev-next" */).then(c => wrapFunctional(c.default || c))
export const LazyPageDocTooltipBtn = import('../../src/components/page/DocTooltipBtn.vue' /* webpackChunkName: "components/page-doc-tooltip-btn" */).then(c => wrapFunctional(c.default || c))
