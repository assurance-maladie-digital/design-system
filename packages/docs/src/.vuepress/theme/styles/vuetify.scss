// Override some Vuetify styles

// Reset max-width for Vuetify ripples
.v-ripple__container,
.v-ripple__container *,
.v-tabs__container,
.v-input--selection-controls__ripple,
.v-slide-group__content,
.v-simple-checkbox svg,
.no-max-width {
	max-width: none;
}

// Fix color contrast
.theme--light.v-input--is-disabled .v-label,
.theme--light.v-icon.v-icon--disabled,
.theme--light.v-list .v-list--disabled {
	color: #767676 !important
}

.theme--light.v-btn.v-btn--disabled .v-icon {
	color: #949494 !important
}

.v-tabs-bar .v-tabs__item--disabled {
	color: #333 !important
}

.theme--light.v-btn.v-btn--disabled {
	color: #646464 !important
}

// Reset Vuetify font-size
.v-btn,
.v-tabs__div,
.body-1 {
	font-size: .85rem !important;
}

table.v-table tbody td,
table.v-table thead th,
.v-datatable__actions,
.body-2 {
	font-size: .8rem;
}

.v-input,
.v-label,
.v-list__tile,
.subtitle-1 {
	font-size: 1rem !important;
}

.v-application {
	.display-4,
	.display-3,
	.display-2,
	.display-1,
	.headline,
	.title,
	.subtitle-1,
	.subtitle-2,
	.body-1,
	.body-2,
	.caption,
	.overline {
		font-family: 'Open Sans', sans-serif !important;
	}
}

.v-data-table__empty-wrapper {
	color: #717171 !important;
}
