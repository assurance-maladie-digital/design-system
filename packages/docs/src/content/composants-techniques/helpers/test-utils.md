---
title: testUtils
description: Le helper `testUtils` contient un ensemble de fonctions utilitaires pour les tests unitaires.
---

<doc-tabs>

<doc-tab-item label="Utilisation">

<doc-alert type="info">

Le helper `testUtils` exporte également toutes les fonctions du package [Vue Test Utils](https://vue-test-utils.vuejs.org/).

</doc-alert>

### applyGlobalConfig

La fonction `applyGlobalConfig` permet d’appliquer la configuration globale pour les tests, elle est utilisée dans le fichier `setup.ts`.

```ts
import { applyGlobalConfig } from '@cnamts/vue-dot/src/helpers/testUtils';

applyGlobalConfig();
```

### installGlobalPlugins

La fonction `installGlobalPlugins` permet d’installer les plugins [Vue Input Facade](https://ronaldjerez.github.io/vue-input-facade/), [Vue Meta](https://vue-meta.nuxtjs.org/), [Vuex](https://vuex.vuejs.org/), [Vuetify](https://v2.vuetifyjs.com/) et Vue Dot pour les tests.

```ts
import { createLocalVue, installGlobalPlugins } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installGlobalPlugins(localVue);
```

### installRouter

La fonction `installRouter` permet d’installer [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createLocalVue, installRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const localVue = createLocalVue();

installRouter(localVue);
```

### createRouter

La fonction `createRouter` permet de créer une instance de [Vue Router](https://v3.router.vuejs.org/fr/) pour les tests.

```ts
import { createRouter } from '@cnamts/vue-dot/src/helpers/testUtils';

const routes = [];

const router = createRouter({
	routes
});
```

### createStore

La fonction `createStore` permet de créer une instance de [Vuex](https://vuex.vuejs.org/) pour les tests.

```ts
import { createStore } from '@cnamts/vue-dot/src/helpers/testUtils';

const store = createStore({
	state: {}
});
```

### createVuetifyInstance

La fonction `createVuetifyInstance` permet de créer une instance de [Vuetify](https://v2.vuetifyjs.com/) pour les tests.

```ts
import { createVuetifyInstance } from '@cnamts/vue-dot/src/helpers/testUtils';

const vuetify = createVuetifyInstance();
```

### mockVFormRef

La fonction `mockVFormRef` permet de mocker la référence d’un composant [VForm](https://v2.vuetifyjs.com/en/components/forms/) pour les tests.

```ts
import { mockVFormRef } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	$refs: {
		form: mockVFormRef(true)
	}
};
```

### mockTranslations

La fonction `mockTranslations` permet de mocker les traductions pour les tests.

```ts
import { mockTranslations } from '@cnamts/vue-dot/src/helpers/testUtils';

const mocks = {
	...mockTranslations<string | string[]>({
		'path.to.title': 'Titre',
		'path.to.items': [
			'Élément 1',
			'Élément 2',
			'Élément 3'
		]
	})
};
```

</doc-tab-item>

<doc-tab-item label="API">
<doc-api name="helpers/test-utils"></doc-api>
</doc-tab-item>

</doc-tabs>
