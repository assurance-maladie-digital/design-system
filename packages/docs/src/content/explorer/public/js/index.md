Le dossier `js` contient les scripts statiques et les variables d'environnement de l'application.

<doc-alert type="warning">
Il n'est pas recommand√© de renommer ce dossier.
</doc-alert>
