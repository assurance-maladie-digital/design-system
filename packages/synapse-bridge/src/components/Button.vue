<script setup lang="ts">
defineProps<{
    text: string
}>()
</script>
<template>
    <button class="button">{{ text }}</button>
</template>

<style scoped></style>
