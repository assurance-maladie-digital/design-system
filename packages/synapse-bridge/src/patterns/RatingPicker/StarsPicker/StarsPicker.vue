<script lang="ts">
import { defineComponent } from "vue";
import { RatingMixin } from '../RatingMixin';
import { mdiStarOutline, mdiStar } from '@mdi/js';
export default defineComponent({
	mixins: [RatingMixin],
	emits: ["update:modelValue"],
	props: {
		length: {
			type: Number,
			default: 5
		}
	},
	data() {
		return {
			starOutlineIcon: mdiStarOutline,
			starIcon: mdiStar,
			hoverIndex: -1
		};
	},
	computed: {
		hasAnswered(): boolean {
			return this.modelValue !== -1;
		}
	},
	methods: {
		isActive(index: number): boolean {
			return this.modelValue - 1 === index;
		},
		isFilled(index: number): boolean {
			const isHovered = this.hoverIndex >= index;
			const isActive = this.modelValue - 1 >= index;
			return isHovered || isActive;
		}
	}
});
</script>

<template>
	<fieldset class="vd-stars-picker">
		<legend class="text-h6 mb-6">
			<slot name="label">
				{{ label }}
			</slot>
		</legend>

		<VRating
			:modelValue="modelValue"
			:length="length"
			:readonly="readonly || hasAnswered"
			class="d-flex flex-wrap max-width-none mx-n3"
			@update:modelValue="emitInputEvent"
		>
			<template #item="{ index }">
				<!-- Using click event on VIcon will convert it into a button -->
				<button
					@mouseover="hoverIndex = index"
					@focus="hoverIndex = index"
					@mouseleave="hoverIndex = -1"
					@blur="hoverIndex = -1"
					:disabled="readonly"
					:aria-pressed="isActive(index)"
				>
					<VIcon
						:class="isFilled(index) ? 'text-am-blue' : 'text-am-blue-lighten-60'"
						size="36px"
						class="py-0 px-2"
					>
						{{ isFilled(index) ? starIcon : starOutlineIcon }}
					</VIcon>
				</button>
			</template>
		</VRating>
	</fieldset>
</template>

<style lang="scss" scoped>
@import '@cnamts/design-tokens/dist/tokens';
.vd-stars-picker {
	border: 0;
}
.v-rating .v-icon {
	width: 52px !important;
	height: 36px !important;
	&--disabled.am-blue--text {
		color: $vd-am-blue-base !important;
	}
	&--disabled.am-blue-lighten-60--text {
		color: $vd-am-blue-lighten-60 !important;
	}
}
</style>
