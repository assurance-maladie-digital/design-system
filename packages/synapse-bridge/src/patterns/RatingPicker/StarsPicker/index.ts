import StarsPicker from './StarsPicker.vue';

export default StarsPicker;
