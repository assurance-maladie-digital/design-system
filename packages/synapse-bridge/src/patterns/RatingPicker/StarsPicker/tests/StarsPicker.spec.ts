import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import { vuetify } from "@tests/unit/setup";

import StarsPicker from "../";
import { VIcon } from "vuetify/components";
import { mdiStarOutline, mdiStar } from '@mdi/js';

describe("StarsPicker", () => {
	it("renders correctly", () => {
		const wrapper = mount(StarsPicker, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				label: "Pourriez-vous donner une note ?",
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it("emits an event when a number is selected", async () => {
		const wrapper = mount(StarsPicker, {
			global: {
				plugins: [vuetify],
			},
		});

		await wrapper.findAll("button")!.at(3)!.trigger("click");

		expect(wrapper.emitted("update:modelValue")).toHaveLength(1);
		expect(wrapper.emitted("update:modelValue")![0]).toEqual([4]);
	});

	it("change the displayed value when the modelValue is updated", async () => {
		const wrapper = mount(StarsPicker, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				modelValue: 3,
			},
		});

		await wrapper.setProps({ modelValue: 4 });
		expect(wrapper.findAll("button")[3].attributes("aria-pressed")).toBe('true');
		await wrapper.setProps({ modelValue: 1 });
		expect(wrapper.findAll("button")[3].attributes("aria-pressed")).toBe('false');
		expect(wrapper.findAll("button")[0].attributes("aria-pressed")).toBe('true');
	});

	it("change the style of the stars on hover", async () => {
		const wrapper = mount(StarsPicker, {
			global: {
				plugins: [vuetify],
			},
		});

		const buttons = wrapper.findAll("button");
		await buttons.at(3)!.trigger("mouseover");

		const icon = buttons!.at(3)!.findComponent(VIcon);
		let slotContent = icon.vm.$slots.default!()![0].children;
		expect(slotContent).toContain(mdiStar);

		await buttons.at(3)!.trigger("mouseleave");
		slotContent = icon.vm.$slots.default!()![0].children;
		expect(slotContent).toContain(mdiStarOutline);
	});

	it("change the style of the stars on focus", async () => {
		const wrapper = mount(StarsPicker, {
			global: {
				plugins: [vuetify],
			},
		});

		const buttons = wrapper.findAll("button");
		await buttons.at(3)!.trigger("focus");

		const icon = buttons!.at(3)!.findComponent(VIcon);
		let slotContent = icon.vm.$slots.default!()![0].children;
		expect(slotContent).toContain(mdiStar);

		await buttons.at(3)!.trigger("blur");
		slotContent = icon.vm.$slots.default!()![0].children;
		expect(slotContent).toContain(mdiStarOutline);
	});
});
