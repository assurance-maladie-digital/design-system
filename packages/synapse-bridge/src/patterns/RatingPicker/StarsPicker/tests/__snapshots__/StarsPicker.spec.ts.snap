// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StarsPicker > renders correctly 1`] = `
<fieldset data-v-e2cb1ead="" class="vd-stars-picker">
  <legend data-v-e2cb1ead="" class="text-h6 mb-6">Pourriez-vous donner une note ?</legend>
  <div data-v-e2cb1ead="" class="v-rating v-theme--light d-flex flex-wrap max-width-none mx-n3"><label for="v-rating-0-0" class=""><span class="v-rating__hidden">Rating 0 of 5</span>
      <!---->
    </label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-0" type="radio" tabindex="-1" value="0">
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-1" class=""><span class="v-rating__hidden">Rating 1 of 5</span><!-- Using click event on VIcon will convert it into a button --><button data-v-e2cb1ead="" aria-pressed="false"><i data-v-e2cb1ead="" class="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z mdi v-icon notranslate v-theme--light text-am-blue-lighten-60 py-0 px-2" style="font-size: 36px; height: 36px; width: 36px;" aria-hidden="true"></i></button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-1" type="radio" tabindex="-1" value="1"></div>
      <!---->
    </div>
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-2" class=""><span class="v-rating__hidden">Rating 2 of 5</span><!-- Using click event on VIcon will convert it into a button --><button data-v-e2cb1ead="" aria-pressed="false"><i data-v-e2cb1ead="" class="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z mdi v-icon notranslate v-theme--light text-am-blue-lighten-60 py-0 px-2" style="font-size: 36px; height: 36px; width: 36px;" aria-hidden="true"></i></button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-2" type="radio" tabindex="-1" value="2"></div>
      <!---->
    </div>
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-3" class=""><span class="v-rating__hidden">Rating 3 of 5</span><!-- Using click event on VIcon will convert it into a button --><button data-v-e2cb1ead="" aria-pressed="false"><i data-v-e2cb1ead="" class="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z mdi v-icon notranslate v-theme--light text-am-blue-lighten-60 py-0 px-2" style="font-size: 36px; height: 36px; width: 36px;" aria-hidden="true"></i></button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-3" type="radio" tabindex="-1" value="3"></div>
      <!---->
    </div>
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-4" class=""><span class="v-rating__hidden">Rating 4 of 5</span><!-- Using click event on VIcon will convert it into a button --><button data-v-e2cb1ead="" aria-pressed="false"><i data-v-e2cb1ead="" class="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z mdi v-icon notranslate v-theme--light text-am-blue-lighten-60 py-0 px-2" style="font-size: 36px; height: 36px; width: 36px;" aria-hidden="true"></i></button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-4" type="radio" tabindex="-1" value="4"></div>
      <!---->
    </div>
    <div class="v-rating__wrapper">
      <!---->
      <div class="v-rating__item"><label for="v-rating-0-5" class=""><span class="v-rating__hidden">Rating 5 of 5</span><!-- Using click event on VIcon will convert it into a button --><button data-v-e2cb1ead="" aria-pressed="false"><i data-v-e2cb1ead="" class="M12,15.39L8.24,17.66L9.23,13.38L5.91,10.5L10.29,10.13L12,6.09L13.71,10.13L18.09,10.5L14.77,13.38L15.76,17.66M22,9.24L14.81,8.63L12,2L9.19,8.63L2,9.24L7.45,13.97L5.82,21L12,17.27L18.18,21L16.54,13.97L22,9.24Z mdi v-icon notranslate v-theme--light text-am-blue-lighten-60 py-0 px-2" style="font-size: 36px; height: 36px; width: 36px;" aria-hidden="true"></i></button></label><input class="v-rating__hidden" name="v-rating-0" id="v-rating-0-5" type="radio" tabindex="-1" value="5"></div>
      <!---->
    </div>
  </div>
</fieldset>
`;
